### ISSUE TITLE
A lot of bugs in primitive ftpd

### ISSUE THREAD

**Issue by venetin** [wolpi/prim-ftpd#331]
Hello,

I am using primitive ftpd since a long time ago. But it started to be unusable with every new version. Here are some of the bugs:
1. In version 7 was introduced a bug when the app asks for a folder on every startup if anu of the SAF methods are selected.
2. The only working method is SAF so I can't select e.g plain old filesystem.
3. If I select the plain old filesystem the folder is emty on the client side. I can see in the logs that when the client is trying to list the contents of the FTP folder for some reason the folder is changed to / which is obviously inaccessible. I am using Nautilus under Ubuntu as a client. I am not sure why this is happening but I am pretty sure that the FTP server must not allow folder browsing outside of the FTP server's root folder
4. In SAF mode there are constant errors when I browse the FTP folder. I need to push F5 after opening each folder in Nautilus so the content will become visible.
The main reason for using this app is that this is probably the only server that can advertise itself in the LAN. But now it is unusable. I switched temporary to the FTP server functionality of the x-plore file manager. It is working really fast and tbere is even an option to serve mofe than one folder. But I will be happy to have primitive ftpd working again. It is more usable on a random networks where there is no way to set static IPs and hostnames like I did with the x-plore in my home network.

**Comment by wolpi** [comment]
> In version 7 was introduced a bug when the app asks for a folder on every startup if anu of the SAF methods are selected.

No idea where this comes from. Fixes are welcome :wink: 


>The only working method is SAF so I can't select e.g plain old filesystem.
>f I select the plain old filesystem the folder is emty on the client side.  ...

Android restricts access to filesystem since a long time. With new Android versions restrictions get stronger. They invented SAF as an alternative to filesystem which they consider more secure.
See README of this repo for some more info. There are also lots of issues here in this regard.


>In SAF mode there are constant errors when I browse the FTP folder. I need to push F5 after opening each folder in Nautilus so the content will become visible.

It's first time I hear about this. Other clients work fine. If I find time I can have a look at Nautilus.

**Comment by wolpi** [comment]
This app is working fine with Nautilus for me. Your issues might have another cause.

**Comment by WalterFaber** [comment]
Funny, I had the same issue and just started 2 hours ago with the analysis. 
In the meantime, wolpi also worked on it....

In detail, I am talking about this part:

"_If I select the plain old filesystem the folder is emty on the client side._"

I am using Nautilus on Ubuntu 22.04 and Lineagoes 20 (Android 13).

Here is how I was able to fix it:
Read this carefully: [https://github.com/wolpi/prim-ftpd#permission](https://github.com/wolpi/prim-ftpd#permission)
:)
In a nutshell: Install it NOT from Playstore and grant special app permission "All files access".

Finally, thank you wolpi for the really great software and all your efforts!!! I use it really frequent with great pleasure!
